sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Security/Ucfg/UcfgBuilder_Instructions.cs

```diff
@@ -233,32 +215,24 @@ public class Class1
         public void Foo(string s)
         {
             string a;
-
             a = s.ToLower();        // %0 := string.ToLower() [ s ]
                                     // a := __id [ %0 ]
-
             a = (s + s).ToLower();  // %1 := __concat [ s s ]
                                     // %2 := string.ToLower() [ %1 ]
                                     // a := __id [ %2 ]
-
             Bar(s.ToLower());       // %3 := string.ToLower() [ s ]
-                                    // %4 := Namespace.Class1.Bar(string) [ %3 ]
-
-            a = string.IsNullOrEmpty(s);    // %5 := string.IsNullOrEmpty(string) [ s ];
-                                            // a := __id [ const ] // not using %5 because it is not string
-
-            a = A(B(C(s)));         // %6 := Namespace.Class1.C(string) [ s ]
-                                    // B is ignored
-                                    // %7 := Namespace.Class1.A(int) [ const ]
-                                    // a := __id [ %7 ]
-
+                                    // %4 := Namespace.Class1.Bar(string) [ this %3 ]
+            a = string.IsNullOrEmpty(s);    // %5 := string.IsNullOrEmpty(string) [ string s ];
+                                            // a := __id [ %5 ]
+            a = A(B(C(s)));         // %6 := Namespace.Class1.C(string) [ this s ]
+                                    // %7 := Namespace.Class1.B(int) [ this %6 ]
+                                    // %8 := Namespace.Class1.A(int) [ this %7 ]
+                                    // a := __id [ %8 ]
             int x;
-            x = O(s);               // %8 := Namespace.Class1.O(object) [ s ];
-                                    // no assignment is added because O returns int
+            x = O(s);               // %9 := Namespace.Class1.O(object) [ this s ]
+                                    // x := __id [ %9 ]

```

Comment:
Also tests with `this.A()` and `other.A()` (let me know if you need details on what I mean)

Reply:
done